### ISSUE TITLE
When decoding a json string, jsonpickle does not re-initialize the class instance.

### ISSUE THREAD

**Issue by MamoonAhmad** [jsonpickle/jsonpickle#442]
Lets say I have this class 

```
class LineItem:
   def __init__(self):
        self.var1 = 1
```

When I encode this json with JSON pickle it will save all the attributes and class definition. And after encoding, lets say I change the class to following.

```
class LineItem:
   def __init__(self):
        self.var1 = 1
        self.var2 = 2
```
        
Now after changing the line item, if I decode the existing json with JSON pickle. The newly decoded class instance of LineItem will not have attribute var2. So if I try to access it, it will throw an error.

In my opinion, there should be an option to reinitialize the class instance.

**Comment by Theelx** [comment]
I agree that such an option should exist. Hopefully I can get around to implementing it over the summer (I'm finishing up my final exams over the next few days).

That being said, it might be very difficult, and there are a lot of bugs that are a higher priority at the moment. If you'd like it sooner, I'd be happy to review and merge a PR, but I can't promise that I'll be able to do the work myself over the next few months (since I'm taking summer classes and have a full-time research job).

**Comment by davvid** [comment]
Perhaps slightly magical, but I added a test case in #445 that demonstrates how you can accomplish this using existing jsonpickle features.

The basic gist of it is that you can provide a boilerplate `__new__` alongside `self.default_factory = self.__class__` and that'll force jsonpickle into calling your constructor during object initialization.

This does mean that your constructor must function without any arguments, but since you implement `__new__` then in theory you can do all kinds of interception there and rewire things as needed after creating the instance. Not advised, but technically possible.

After construction jsonpickle will fill in any data that was present in the json payload, so if you have json with old attributes those will continue to be present in new instances.

Hopefully this approach works for your use case.

**Comment by Theelx** [comment]
Thanks so much davvid!
